src/sentry/event_manager.py

```diff
@@ -142,7 +143,23 @@ def has_pending_commit_resolution(group):
     if recent_group_link is None:
         return False
 
-    return not ReleaseCommit.objects.filter(commit__id=recent_group_link.linked_id).exists()
+    # commit has been released and is not in pending commit state
+    if ReleaseCommit.objects.filter(commit__id=recent_group_link.linked_id).exists():
+        return False
+    else:
+        # check if this commit is a part of a PR
+        pr_ids = list(
+            PullRequest.objects.filter(
+                pullrequestcommit__commit=recent_group_link.linked_id
+            ).values_list("id", flat=True)
+        )
+        if pr_ids:
+            # assume that this commit has been released if any commits in this PR have been released
+            if ReleaseCommit.objects.filter(
+                commit__pullrequestcommit__pull_request__in=pr_ids
+            ).exists():
```

Comment:
We discussed your concerns about assuming any commits within a PR that were released should regress the issue. 

It might be worthwhile to call it out.

I think it should be fine, but others may disagree.

Reply:
Yeah, we discussed the idea that users may get more noise about regressing issues given our assumption in this specific case (normally a released commit within a PR doesn’t mean all the commits in the PR are also released). Would appreciate your thoughts on this @wedamija

Reply:
So I agree that just because a pr is merged doesn't necessarily mean all of the commits would be released - it's possible that if people aren't using a squash workflow, then merging to main would result in a bunch of new commits. Potentially that could mean that some of the commits wouldn't be deployed, since someone could decide to deploy a sha that doesn't include everything.

In practice though, I think that from my point of view a pr is considered a unit of work, and if it's merged to main I would expect the whole pr to go out in a deploy. I'm sure there is someone out there that is doing something different, but I think that most people will operate this way, so this should be safe.